src/ai/defaultai_seafaring.cc

```diff
@@ -171,15 +173,8 @@ bool DefaultAI::marine_main_decisions(const Time& gametime) {
 		}
 	}
 
-	// iterates over all ships
-	for (const ShipObserver& observer : allships) {
-		if (observer.ship->state_is_expedition()) {
-			++expeditions_in_progress;
-		}
-	}
-
-	assert(allships.size() >= expeditions_in_progress);
-	bool ship_free = allships.size() - expeditions_in_progress > 0;
+	assert(tradeships_count >= expeditions_in_progress);
+	bool ship_free = tradeships_count - expeditions_in_progress - ports_count / 3 > 0;
```

Comment:
bunnybot didn't pick this up from CB. I think you could convert these like this to avoid ugly casts:

```suggestion
	bool ship_free = tradeships_count > expeditions_in_progress + ports_count / 3;
```